packages/editor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -150,19 +143,14 @@ export class PropertyTabComponent extends react.Component<IPropertyTabComponentP
         this.props.globalState.saveSmartFilter();
     }
 
-    customSave() {
-        // this.setState({ uploadInProgress: true });
-        // this.props.globalState.onLogRequiredObservable.notifyObservers(new LogEntry("Saving your material to Babylon.js snippet server...", false));
-        // this.props.globalState
-        //     .customSave!.action(SerializationTools.Serialize(this.props.globalState.nodeMaterial, this.props.globalState))
-        //     .then(() => {
-        //         this.props.globalState.onLogRequiredObservable.notifyObservers(new LogEntry("Material saved successfully", false));
-        //         this.setState({ uploadInProgress: false });
-        //     })
-        //     .catch((err) => {
-        //         this.props.globalState.onLogRequiredObservable.notifyObservers(new LogEntry(err, true));
-        //         this.setState({ uploadInProgress: false });
-        //     });
+    async customSave() {
+        this.setState({ uploadInProgress: true });
+        try {
+            this.props.globalState.onSaveEditorDataRequiredObservable.notifyObservers();
+            await this.props.globalState.customSave();
+        } finally {
+            this.setState({ uploadInProgress: false });
+        }
     }
 
     saveToSnippetServer() {
```

Comment:
How is this used, should we be reusing it?

Reply:
In NME, saveToSnippetServer() is used by the playground version of NME, and customSave() is used in the standalone version. So, no, we don't need both saveToSnippet() and customSave() :)